src/Adapter/ContainerBuilder.php

```diff
@@ -57,16 +62,80 @@ public static function getContainer($name, $isDebug)
         }
 
         $container = new SfContainerBuilder();
+
+        $parameters = require _PS_ROOT_DIR_ . '/app/config/parameters.php';
+        foreach ($parameters['parameters'] as $parameter => $value) {
+            $container->setParameter($parameter, $value);
+        }
+        $env = $isDebug ? 'dev' : 'prod';
+        $container->setParameter('kernel.bundles', []);
+        $container->setParameter('kernel.root_dir', _PS_ROOT_DIR_ . '/app/');
+        $container->setParameter('kernel.name', 'app');
+        $container->setParameter('kernel.debug', $isDebug);
+        $container->setParameter('kernel.environment', $env);
+        $container->setParameter('kernel.cache_dir', _PS_CACHE_DIR_);
+
         $container->addCompilerPass(new LegacyCompilerPass());
+
+        $moduleRepository = ModuleRepositoryFactory::getInstance()->getRepository();
+        if (null !== $moduleRepository) {
+            $activeModules = $moduleRepository->getActiveModules();
+            self::addDoctrine($container, $activeModules);
+            self::enableComposerAutoloaderOnModules($activeModules);
+        }
+
         $loader = new YamlFileLoader($container, new FileLocator(__DIR__));
-        $env = $isDebug ? 'dev' : 'prod';
```

Comment:
uh_uh this line could create some issue when launching tests, like using the real database instead of the fake one: well spotted :+1:

Reply:
so we should manage the test environment? as suggested in your other comment?

Reply:
https://github.com/PrestaShop/modok/blob/master/public/index.php#L18
This is how Symfony manage environement, but the problem is PrestaShop is only checking for `isDebug` or not :/

Reply:
et le isDebug vient de `_PS_MODE_DEV_` qui est paramétrable, je vais rajouter un troisième paramètre et j'utiliserai la méthode de symfony pour le remplir

Reply:
mmm, ou plutôt faire ça `$env = isset($_SERVER['APP_ENV']) ? $_SERVER['APP_ENV'] : ($isDebug ? 'dev' : 'prod');`

Reply:
euh par contre je viens de voir un truc, le dump est mis dans `_PS_CACHE_DIR_` qui dépend de `_PS_MODE_DEV_` mais y'a rien pour gérer dans un env de test, donc ça peut dumper dans le cache de dev..
https://github.com/PrestaShop/PrestaShop/blob/75dda346dd95d489c7ae56aeb1d9ab3f33ac841b/config/defines.inc.php#L72

est-ce que ça vaut le coup de gérer `$_SERVER['APP_ENV']` à ce niveau là aussi?

Reply:
Should be better to have a service PrestaShopEnv (prestashop.env) or something similar to retrieve the current environment. But in a new PR maybe?

Reply:
Yep this would prevent from duplicating code, probably a Core interface with an Adapter
Although another PR would be better I'm afraid it will delay the time to validate/integrate it.. Maybe we could do it in this PR?

Reply:
In a new PR, otherwise, too much code, too much reviews, hard to rebase :smiley:

Reply:
sure, is it ok for the rest of environment management for now? in `ContainerBuilder.php` and `ModuleRepositoryFactory` ?

Reply:
yes, now we know why it's done like that: no problems!